plugins/serializers/prometheusremotewrite/prometheusremotewrite_test.go

```diff
@@ -201,6 +202,98 @@ http_request_duration_seconds_bucket{le="0.5"} 129389
 	}
 }
 
+// fakeLogger immitates telegraf.Logger but preserves
+// the last recorded message.
+type fakeLogger struct {
+	telegraf.Logger
+	lastMsg string
+}
+
+// Errorf overrides telegraf.Logger method to store the message in lastMsg.
+// lastMsg can be then used for testing the output
+func (fl *fakeLogger) Errorf(format string, args ...interface{}) {
+	fl.lastMsg = fmt.Sprintf(format, args...)
+}
+
+func (fl *fakeLogger) has(msg string) bool {
+	return strings.Contains(fl.lastMsg, msg)
+}
+
```

Comment:
Please use [`testutil.CaptureLogger`](https://github.com/influxdata/telegraf/blob/master/testutil/capturelog.go) instead of implementing your own version!

Reply:
Updated in https://github.com/influxdata/telegraf/pull/15893/commits/265065b915a2753a7597139d3886f14e570fd816